liferay-headless/liferay-portal
# LPS-166111 Add SF rules to add the externalReferenceCode to the add methods in the service layer

Related issue: https://issues.liferay.com/browse/LPS-166111

---

This PR contains the code to create the following SF rules:

- `JavaLocalServiceImplErcUsageCheck`: For every `<EntityName>LocalServiceImpl.java` classes whose `<EntityName>` has `externalReferenceCode`:
  - Includes the externalReferenceCode parameter as a parameter in the `add` method of the service layer if the parameter does not exist.
  - Invoke the invocation to the `.setExternalReferenceCode(externalReferenceCode)` method if the method does not contain it.
- `JavaServiceImplErcUsageCheck`:  For every `<EntityName>ServiceImpl.java` classes whose `<EntityName>` has `externalReferenceCode`:
   - Includes the externalReferenceCode parameter as a parameter in the `add` method of the service layer if the parameter does not exist.
  - Include the externalReferenceCode parameter to any invocation to the `add` method.

By default, the SF rules are **deactivated**.

A new parameter has been added to activate the rules: `formatSource.check.erc.usage`.

---

**How to test the SF rules over a module**

For example, for the module that is in the folder `modules/apps/commerce/commerce-inventory-service`:

- `gradlew formatSource -D:apps:commerce:commerce-inventory-service:formatSource.check.erc.usage=true`

---

See https://github.com/ccorreagg/liferay-portal/pull/21

cc/ @ling-alan-huang
